Implement Profile Edit

BikerIndian opened the pull request.

BikerIndian committed:
feat: add profile edit loyout

arsensaliev reviewed (approved):
Ваш верстка выглядит плохо, сделайте так как в figma. После того как это сделаете исправьте конфликты.

arsensaliev reviewed (changes requested):
![Profile Edit (1)](https://user-images.githubusercontent.com/52528182/129443840-9275fe5d-5165-4b33-812d-35f281e927be.png)
![image](https://user-images.githubusercontent.com/52528182/129443852-8f9d56e7-4773-43f3-b0a0-5a50925f4f7d.png)

То что вы сделали похоже на итоговую ?

BikerIndian committed:
feat: corrections on comments

BikerIndian committed:
refactor: backspace button

BikerIndian committed:
feat: corrections on comments

BikerIndian committed:
refactor: moved image to core

arsensaliev commented on feature-profile-edit/src/main/res/layout/fragment_profile_edit.xml:
```diff
@@ -0,0 +1,137 @@
+                android:scaleType="centerCrop"
+                android:src="@drawable/ex_profile_avatar"
+                android:tint="#80000000"
+                tools:ignore="UseAppTint" />
```
Нельзя игнорить ошибки.

arsensaliev replied:
Также если не получается сделать затемнение фото, я могу подсказать один метод. 
Просто засуньте ImageView во FrameLayout. И выставьте FrameLayout:
android:foreground="#80000000"

BikerIndian replied:
затемнение фото работает, засовывать ImageView во FrameLayout?

arsensaliev replied:
Да

arsensaliev replied:
Да

arsensaliev commented on feature-profile-edit/src/main/res/layout/fragment_profile_edit.xml:
```diff
@@ -0,0 +1,137 @@
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toTopOf="parent"
+            tools:ignore="MissingConstraints">
```
Нельзя игнорить ошибки!

BikerIndian replied:
При отключении ошибок Android Studio думает, что
В TextInputLayout не указаны размеры, хотя эти размеры указаны в стилях.
Я отключаю игнорирование ошибок ошибок и дублирую размеры во всех TextInputLayout?

arsensaliev replied:
Не знаю как это сделать, как но всё равно уберите игнорирование ошибки!

BikerIndian replied:
Исправил

RomchaDev replied:
Все ошибки нужно исправить. Touch size too small говорит о том, что нужно увеличить размер editText. TextAppearence мы уже обсуждали

BikerIndian replied:
Нужно поставить в студии Settings > Experimental > [x] Use new Layout Rendering Engine

arsensaliev commented on feature-profile-edit/src/main/res/values/themes.xml:
```diff
@@ -0,0 +1,59 @@
+    <style name="ProfileEditBottom" parent="Widget.MaterialComponents.Button.TextButton">
+        <item name="android:textAppearance">@style/ProfileEditTextStyle</item>
+        <item name="android:textColor">@color/white</item>
+        <item name="android:textSize">14dp</item>
```
Размер текста надо указывать в sp !

BikerIndian replied:
Исправил

arsensaliev commented on feature-profile-edit/src/main/res/values/themes.xml:
```diff
@@ -0,0 +1,59 @@
+        <item name="android:textColor">@color/black</item>
+        <item name="android:layout_width">match_parent</item>
+        <item name="android:layout_height">wrap_content</item>
+        <item name="android:textSize">15dp</item>
```
Размер текста надо указывать в sp !

BikerIndian replied:
Исправил

arsensaliev commented on feature-profile-edit/src/main/res/values/strings.xml:
```diff
@@ -0,0 +1,15 @@
+<resources>
+    <string name="btn_backspace_description">Назад</string>
+    <string name="profile_edit_name">Имя</string>
+    <string name="profile_edit_surname">Фамилия</string>
```
Лучше использовать слова firstName и lastName .

arsensaliev replied:
Лучше вместо surname использовать profile_lastName. Также почему здесь есть слово edit, а в firstname нет ?

BikerIndian replied:
Исправил

arsensaliev commented on feature-profile-edit/src/main/res/values/strings.xml:
```diff
@@ -0,0 +1,15 @@
+    <string name="profile_edit_btn">Сохранить</string>
+
+    <string name="profile_edit_name_err">Заполните имя</string>
+    <string name="profile_edit_surname_err">Заполните фамилию</string>
```
Лучше использовать слова firstName и lastName .

BikerIndian replied:
Сделано

arsensaliev replied:
Нет, не сделано, вы не переименовали surname на last_name.

BikerIndian replied:
переименовал в  last_name

arsensaliev commented:
Нужно исправить конфликты!

BikerIndian committed:
refactor: id name

arsensaliev commented on feature-profile-edit/src/main/res/layout/fragment_profile_edit.xml:
```diff
@@ -0,0 +1,145 @@
+            android:layout_gravity="center"
+            android:layout_marginStart="20dp"
+            android:layout_marginTop="120dp"
+            android:layout_marginEnd="20dp"
```
Размеры укажи в dimens в своем модуле.

BikerIndian replied:
перенес

arsensaliev commented on feature-profile-edit/src/main/res/values/themes.xml:
```diff
@@ -0,0 +1,59 @@
+        <item name="android:textColor">@color/white</item>
+        <item name="android:textSize">14dp</item>
+
+        <item name="android:backgroundTint">#6550EE</item>
```
цвета надо брать с colors ресурса

BikerIndian replied:
перенес

arsensaliev commented on feature-profile-edit/src/main/res/values/themes.xml:
```diff
@@ -0,0 +1,59 @@
+
+        <item name="errorEnabled">true</item>
+        <item name="errorIconTint">@null</item>
+        <item name="boxStrokeColor">#6200EE</item>
```
цвета надо брать с colors ресурса

arsensaliev replied:
Кстати это неправильный цвет. Должен быть как в core модуле. color/purple_light

BikerIndian replied:
сделано

arsensaliev commented on feature-profile-edit/src/main/res/layout/fragment_profile_edit.xml:
```diff
@@ -0,0 +1,145 @@
+                android:contentDescription="@string/profile_avatar"
+                android:scaleType="centerCrop"
+                android:src="@drawable/ex_profile_avatar"
+                android:tint="#80000000"
```
цвета надо брать с colors ресурса и указывать в themes лучше указывать стили.

BikerIndian replied:
перенес в colors

arsensaliev reviewed (changes requested):
Контент должен быть ровно по центру экрана. Она сейчас выше чем надо. Также почему аватарка в отдельном CardView ? Надо было весь контент в один контейнер засунуть.
![image](https://user-images.githubusercontent.com/52528182/129890929-18f615d5-7165-4efb-a7c9-99db95348a2a.png)

arsensaliev commented on feature-profile-edit/src/main/res/drawable/ic_camera.xml:
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="utf-8"?>
+<layer-list xmlns:android="http://schemas.android.com/apk/res/android">
```
Зачем этот файл ? Думаю этот файл бесполезный, так как можно сразу указать на @drawble/ic_bx_camera.

arsensaliev replied:
Зачем без разрешения нажали на кнопку Resolve conversation ?

BikerIndian replied:
файл убрал

arsensaliev commented on feature-profile-edit/src/main/java/com/demal/feature_profile_edit/view/ProfileEditFragment.kt:
```diff
@@ -0,0 +1,7 @@
+import com.demal.feature_profile_edit.R
+
+//Только для теста
+class ProfileEditFragment : Fragment(R.layout.fragment_profile_edit)
```
Надо наследоваться от BaseFragment

BikerIndian replied:
добавил наследование

BikerIndian committed:
feat: corrections on comments
+ проверка на пустые поля ввода
+ ScrollView
+ Центровка окна
+ id/profile_edit_name_layout > profile_edit_firstName_layout

arsensaliev commented on feature-profile-edit/src/main/java/com/demal/feature_profile_edit/view/ProfileEditFragment.kt:
```diff
@@ -0,0 +1,78 @@
+
+        val editCityLayout = view.findViewById<TextInputLayout>(R.id.profile_edit_сity_layout)
+        val editCity = view.findViewById<TextInputEditText>(R.id.profile_edit_сity)
+        val сityMessageError = resources.getString(R.string.profile_edit_сity_err)
```
Мы решили в проекте использовать ViewBinding, уберите весь этот код.

BikerIndian replied:
добавил Binding

RomchaDev replied:
bindingNullable для других целей создан. Используй binding

BikerIndian replied:
сделал

arsensaliev commented on feature-profile-edit/src/main/java/com/demal/feature_profile_edit/view/ProfileEditFragment.kt:
```diff
@@ -0,0 +1,78 @@
+import com.google.android.material.textfield.TextInputLayout
+
+//Только для теста
+class ProfileEditFragment : Fragment(R.layout.fragment_profile_edit) {
```
Надо наследоваться от BaseFragment в core модуле

BikerIndian replied:
добавил наследование

arsensaliev commented on feature-profile-edit/src/main/java/com/demal/feature_profile_edit/view/ProfileEditFragment.kt:
```diff
@@ -0,0 +1,78 @@
+//Только для теста
+class ProfileEditFragment : Fragment(R.layout.fragment_profile_edit) {
+
+    private val EMPTY_STRING = ""
```
Зачем нужна эта переменная ?

BikerIndian replied:
Потому что в коде не должно быть текст указан в явном виде. Поменял на null как показано в material.io

arsensaliev commented on feature-profile-edit/src/main/java/com/demal/feature_profile_edit/view/ProfileEditFragment.kt:
```diff
@@ -0,0 +1,78 @@
+
+    }
+
+    private fun setEditorListener(editFirstNameLayout: TextInputLayout, editFirstName: TextInputEditText, firstNameMessageError: String) {
```
В параметрах указан editFirstName но вы в других местах передаёте country и city и surName. Тогда стоит переименовать параметр на просто inputLayout.

BikerIndian replied:
переименовал

arsensaliev commented on feature-profile-edit/src/main/java/com/demal/feature_profile_edit/view/ProfileEditFragment.kt:
```diff
@@ -0,0 +1,78 @@
+        }
+    }
+
+    private fun checkErrors(inputLayout: TextInputLayout, editText: TextInputEditText, firstNameMessageError: String): Boolean {
```
Причем тут параметр firstNameMessageError ? Она же не отображает ошибку только для имени! По коду видно что это универсальный метод отображение ошибок. Стоит переименовать на errorMessage!

BikerIndian replied:
переименовал

arsensaliev commented on feature-profile-edit/src/main/java/com/demal/feature_profile_edit/view/ProfileEditFragment.kt:
```diff
@@ -0,0 +1,78 @@
+    }
+
+    private fun checkErrors(inputLayout: TextInputLayout, editText: TextInputEditText, firstNameMessageError: String): Boolean {
+        return if (shouldShowError(editText)) {
```
Так лучше не возвращать значения. Такой стиль возврата значения путает разработчика. Напишите обычный стиль if else и внутри каждого возвращайте значение

arsensaliev replied:
ПОЖАЛУЙСТА не нажимайте кнопку resolve conversation. Это только должен делать reviewer!  У вас есть причина нажатия ?

BikerIndian replied:
fixed: На митинге мне сказал что могу закрыть так как этот код убран.

arsensaliev commented on feature-profile-edit/src/main/java/com/demal/feature_profile_edit/view/ProfileEditFragment.kt:
```diff
@@ -0,0 +1,78 @@
+        }
+    }
+
+    private fun showErrorOn(editText: TextInputLayout, errText: String) {
```
В некоторых местах у вас inputLayout, здесь почему то editText. Может выберите одно ? Например inputLayout.

arsensaliev replied:
Также не стоит звать параметры коротки названиями. Назови его более понятным, например errorMessage.

BikerIndian replied:
проверка будет из библиотеки

arsensaliev commented on feature-profile-edit/src/main/java/com/demal/feature_profile_edit/view/ProfileEditFragment.kt:
```diff
@@ -0,0 +1,78 @@
+        editText.error = errText
+    }
+
+    private fun showErrorOff(editText: TextInputLayout) {
```
В некоторых местах у вас inputLayout, здесь почему то editText. Может выберите одно ? Например inputLayout.

arsensaliev replied:
ПОЖАЛУЙСТА не нажимайте кнопку resolve conversation. Это только должен делать reviewer! У вас есть причина нажатия ?

BikerIndian replied:
fixed: На митинге мне сказал что могу закрыть так как этот код убран.

arsensaliev commented on feature-profile-edit/src/main/java/com/demal/feature_profile_edit/view/ProfileEditFragment.kt:
```diff
@@ -0,0 +1,78 @@
+
+    private fun shouldShowError(editText: TextInputEditText): Boolean {
+        if (editText.text != null) {
+            val textLength = editText.text!!.length
```
Перед return лучше оставлять отступ.

arsensaliev replied:
ПОЖАЛУЙСТА не нажимайте кнопку resolve conversation. Это только должен делать reviewer! У вас есть причина нажатия ?

BikerIndian replied:
fixed: На митинге мне сказал что могу закрыть так как этот код убран.

arsensaliev commented on feature-profile-edit/src/main/java/com/demal/feature_profile_edit/view/ProfileEditFragment.kt:
```diff
@@ -0,0 +1,78 @@
+            val textLength = editText.text!!.length
+            return textLength > 0
+        }
+        return false
```
Перед return лучше оставлять отступ.

BikerIndian replied:
fixed:  этот код удален.

arsensaliev commented on feature-profile-edit/src/main/java/com/demal/feature_profile_edit/view/ProfileEditFragment.kt:
```diff
@@ -0,0 +1,78 @@
+
+
+
+
```
Удалите лишние отступы.

BikerIndian replied:
убрал

arsensaliev commented on feature-profile-edit/src/main/res/layout/fragment_profile_edit.xml:
```diff
@@ -0,0 +1,154 @@
+        android:id="@+id/scrollView">
+    <androidx.constraintlayout.widget.ConstraintLayout
+        android:layout_width="match_parent"
+        android:layout_height="match_parent"
```
Студия показывает здесь предупреждение. Все предупреждения стоит исправить!

BikerIndian replied:
исправлено

arsensaliev commented on feature-profile-edit/src/main/res/layout/fragment_profile_edit.xml:
```diff
@@ -0,0 +1,154 @@
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent">
+    <ScrollView
```
Добавьте отступ перед каждый новым view

BikerIndian replied:
добавил

arsensaliev commented on feature-profile-edit/src/main/res/layout/fragment_profile_edit.xml:
```diff
@@ -0,0 +1,154 @@
+        android:layout_height="match_parent"
+        android:fillViewport="true"
+        android:id="@+id/scrollView">
+    <androidx.constraintlayout.widget.ConstraintLayout
```
Добавьте отступ перед каждый новым view

BikerIndian replied:
добавил

arsensaliev commented on feature-profile-edit/src/main/res/layout/fragment_profile_edit.xml:
```diff
@@ -0,0 +1,154 @@
+        android:background="@drawable/profile_bg_dark">
+
+
+
```
Здесь слишком много отступов. Хватит и одного

BikerIndian replied:
fixed

arsensaliev commented on feature-profile-edit/src/main/res/layout/fragment_profile_edit.xml:
```diff
@@ -0,0 +1,154 @@
+            android:id="@+id/profile_edit_btn_backspace"
+            android:layout_width="34dp"
+            android:layout_height="34dp"
+            android:layout_marginStart="20dp"
```
Все размеры надо указывать в dimens!

BikerIndian replied:
указал

arsensaliev commented on feature-profile-edit/src/main/res/layout/fragment_profile_edit.xml:
```diff
@@ -0,0 +1,154 @@
+                    <com.google.android.material.textfield.TextInputEditText
+                        android:id="@+id/profile_edit_firstName"
+                        style="@style/profileEditeditText" />
+                </com.google.android.material.textfield.TextInputLayout>
```
Добавьте отступ перед закрывающим TextInputLayout.

BikerIndian replied:
добавил

arsensaliev commented on feature-profile-edit/src/main/res/layout/fragment_profile_edit.xml:
```diff
@@ -0,0 +1,154 @@
+                <com.google.android.material.textfield.TextInputLayout
+                    android:id="@+id/profile_edit_firstName_layout"
+                    style="@style/profileEdittextInputLayout"
+                    android:layout_marginTop="86dp"
```
Размеры в dimens

BikerIndian replied:
убрал

arsensaliev commented on feature-profile-edit/src/main/res/layout/fragment_profile_edit.xml:
```diff
@@ -0,0 +1,154 @@
+                    <com.google.android.material.textfield.TextInputEditText
+                        android:id="@+id/profile_edit_surname"
+                        style="@style/profileEditeditText" />
+                </com.google.android.material.textfield.TextInputLayout>
```
Добавьте отступ перед закрывающим TextInputLayout.

BikerIndian replied:
добавил

arsensaliev commented on feature-profile-edit/src/main/res/layout/fragment_profile_edit.xml:
```diff
@@ -0,0 +1,154 @@
+                    <com.google.android.material.textfield.TextInputEditText
+                        android:id="@+id/profile_edit_сountry"
+                        style="@style/profileEditeditText" />
+                </com.google.android.material.textfield.TextInputLayout>
```
Добавьте отступ перед закрывающим TextInputLayout.

BikerIndian replied:
fixed

arsensaliev commented on feature-profile-edit/src/main/res/layout/fragment_profile_edit.xml:
```diff
@@ -0,0 +1,154 @@
+                    <com.google.android.material.textfield.TextInputEditText
+                        android:id="@+id/profile_edit_сity"
+                        style="@style/profileEditeditText" />
+                </com.google.android.material.textfield.TextInputLayout>
```
Добавьте отступ перед закрывающим TextInputLayout.

BikerIndian replied:
добавил

arsensaliev commented on feature-profile-edit/src/main/res/layout/fragment_profile_edit.xml:
```diff
@@ -0,0 +1,154 @@
+                    android:id="@+id/profile_edit_сity_layout"
+                    style="@style/profileEdittextInputLayout"
+                    android:hint="@string/profile_edit_сity"
+                    app:hintAnimationEnabled="false"
```
Зачем вы эту опцию добавили ? Почему она нет в остальных ?

BikerIndian replied:
убрал

BikerIndian replied:
был для теста

arsensaliev commented on feature-profile-edit/src/main/res/layout/fragment_profile_edit.xml:
```diff
@@ -0,0 +1,154 @@
+                    android:id="@+id/profile_edit_btn_send"
+                    style="@style/ProfileEditButton"
+                    android:layout_marginTop="0dp"
+                    android:layout_marginBottom="20dp"
```
В dimens укажите размеры

BikerIndian replied:
указал

arsensaliev commented on feature-profile-edit/src/main/res/layout/fragment_profile_edit.xml:
```diff
@@ -0,0 +1,154 @@
+
+            </androidx.constraintlayout.widget.ConstraintLayout>
+
+
```
Здесь 2 отступа. Хватит и одного.

BikerIndian replied:
сделал

arsensaliev commented on feature-profile-edit/src/main/res/layout/fragment_profile_edit.xml:
```diff
@@ -0,0 +1,154 @@
+
+        </androidx.cardview.widget.CardView>
+
+        <androidx.cardview.widget.CardView
```
Я говорил чтобы всё это было в одном контейнере. Так как в дизайне видно что всё находится в одном контейнере. 
![image](https://user-images.githubusercontent.com/52528182/130180828-9df1384c-cdb2-493b-acc4-8c04c43939f0.png)

BikerIndian replied:
fixed

arsensaliev commented on feature-profile-edit/src/main/res/layout/fragment_profile_edit.xml:
```diff
@@ -0,0 +1,154 @@
+        <androidx.cardview.widget.CardView
+            android:id="@+id/cardView"
+            style="@style/profileEditPhoto"
+
```
Уберите лишний отступ.

BikerIndian replied:
fixed

arsensaliev commented on feature-profile-edit/src/main/res/layout/fragment_profile_edit.xml:
```diff
@@ -0,0 +1,154 @@
+
+            <ImageView
+                android:id="@+id/profile_edit_img_avatar"
+                android:layout_width="91dp"
```
в dimens значение поставьте.

BikerIndian replied:
fixed

arsensaliev commented on feature-profile-edit/src/main/res/layout/fragment_profile_edit.xml:
```diff
@@ -0,0 +1,154 @@
+        </androidx.cardview.widget.CardView>
+
+
+
```
Излишние отступы.

BikerIndian replied:
fixed

arsensaliev commented on feature-profile-edit/src/main/res/layout/fragment_profile_edit.xml:
```diff
@@ -0,0 +1,154 @@
+                android:scaleType="centerCrop"
+                android:src="@drawable/ic_bx_camera" />
+
+
```
Излишние отступы.

BikerIndian replied:
убрал

BikerIndian replied:
fixed

arsensaliev commented on feature-profile-edit/src/main/res/layout/fragment_profile_edit.xml:
```diff
@@ -0,0 +1,154 @@
+
+
+    </androidx.constraintlayout.widget.ConstraintLayout>
+    </ScrollView>
```
Добавьте 1 отступ перед каждым новым view.

BikerIndian replied:
сделано

arsensaliev commented on feature-profile-edit/src/main/res/values/strings.xml:
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+    <string name="btn_backspace_description">Назад</string>
+    <string name="profile_firstName_name">Имя</string>
```
Почему слово name повторяется ? profile_first_name можете назвать.

BikerIndian replied:
убрал

arsensaliev commented on feature-profile-edit/src/main/res/values/themes.xml:
```diff
@@ -0,0 +1,57 @@
+        <item name="cardCornerRadius">70dp</item>
+    </style>
+
+
```
излишний отступ. Также все размеры которые здесь укажите в dimens.

BikerIndian replied:
отступы убрал

arsensaliev reviewed (changes requested):
Исправьте конфликты.

BikerIndian committed:
Merge remote-tracking branch 'origin/dev' into feature/profile-edit

# Conflicts:
#	app/build.gradle
#	buildSrc/src/main/java/Dependencies.kt
#	feature-profile/src/main/java/com/demal/feature_profile/view/ProfileFragment.kt
#	settings.gradle

BikerIndian committed:
merge: dev > feature-profile-edit

BikerIndian committed:
refactor: : corrections on comments

BikerIndian committed:
refactor: : corrections on comments

BikerIndian committed:
feat: add profile edit in app

BikerIndian committed:
refactor: set margin in button backspace

BikerIndian committed:
feat: add EditText Validator

BikerIndian committed:
feat: button backspace no scroll

BikerIndian committed:
refactor: corrections on comments (binding)

BikerIndian committed:
feat: add validation library

BikerIndian committed:
feat: add to Screens

BikerIndian committed:
Merge remote-tracking branch 'origin/dev' into feature/profile-edit

# Conflicts:
#	app/src/main/java/com/demal/koin/modules/ViewModelModules.kt

BikerIndian committed:
refactor: corrections

arsensaliev commented on feature-profile-edit/build.gradle:
```diff
@@ -0,0 +1,26 @@
+    implementation Design.appcompat
+    implementation Design.material
+    implementation Design.constraintLayout
+    implementation 'com.basgeekball:awesome-validation:4.3'
```
Нужно указать в Dependencies

BikerIndian replied:
Так на митинге договорились что этого не делаю.

arsensaliev commented on feature-profile-edit/src/main/java/com/demal/feature_profile_edit/main/ProfileEditFragment.kt:
```diff
@@ -0,0 +1,92 @@
+class ProfileEditFragment :
+    BaseFragment<FragmentProfileEditBinding, User, ProfileEditViewModel>() {
+
+    private var mAwesomeValidation = AwesomeValidation(ValidationStyle.TEXT_INPUT_LAYOUT)
```
Переменную лучше назвать validator

BikerIndian replied:
переименовал

arsensaliev commented on feature-profile-edit/src/main/java/com/demal/feature_profile_edit/main/ProfileEditFragment.kt:
```diff
@@ -0,0 +1,92 @@
+
+    override fun onStart() {
+        super.onStart()
+        viewModel.init()
```
viewModel должен инициализироваться в onViewCreated

BikerIndian replied:
эта часть кода из ProfileFragment

BikerIndian replied:
перенес

arsensaliev commented on feature-profile-edit/src/main/java/com/demal/feature_profile_edit/main/ProfileEditViewModel.kt:
```diff
@@ -0,0 +1,15 @@
+    navigator: ProfileEditNavigator,
+    private val repository: UserRepository
+) : BaseViewModel<User>(navigator) {
+    fun init() {
```
Если данный метод пустой, то лучше удалить

BikerIndian replied:
удалил

arsensaliev commented on feature-profile-edit/src/main/java/com/demal/feature_profile_edit/navigation/ProfileEditNavigator.kt:
```diff
@@ -0,0 +1,8 @@
+
+import com.demal.view.core.BaseNavigator
+
+interface ProfileEditNavigator : BaseNavigator {
```
У вас вроде неправильно сделана навигация, спросите у Романа как должна быть.

arsensaliev commented on feature-profile-edit/src/main/res/values/colors.xml:
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+    <color name="fogging_80">#80000000</color>
+    <color name="button_save_color">#6550EE</color>
```
Данные цвета есть в core. Удалите этот файл.

BikerIndian replied:
удалил

arsensaliev commented on feature-profile-edit/src/main/res/values/dimens.xml:
```diff
@@ -0,0 +1,24 @@
+    <dimen name="margin_top_first_edit_text_first_name">86dp</dimen>
+    <dimen name="edit_text_size_text">20sp</dimen>
+    <dimen name="edit_text_layout_height">85sp</dimen>
+    <dimen name="edit_text_layout_margin_start">20sp</dimen>
```
Должен быть вроде в dp. Так как в sp указываем размер текста.

BikerIndian replied:
исправил

arsensaliev commented on feature-profile-edit/src/main/res/values/dimens.xml:
```diff
@@ -0,0 +1,24 @@
+    <dimen name="edit_text_size_text">20sp</dimen>
+    <dimen name="edit_text_layout_height">85sp</dimen>
+    <dimen name="edit_text_layout_margin_start">20sp</dimen>
+    <dimen name="edit_text_layout_marginEnd">20sp</dimen>
```
Должен быть вроде в dp. Так как в sp указываем размер текста.

BikerIndian replied:
исправил

arsensaliev commented on feature-profile-edit/src/main/res/values/dimens.xml:
```diff
@@ -0,0 +1,24 @@
+    <dimen name="radius_profile_edit">25dp</dimen>
+    <dimen name="margin_top_first_edit_text_first_name">86dp</dimen>
+    <dimen name="edit_text_size_text">20sp</dimen>
+    <dimen name="edit_text_layout_height">85sp</dimen>
```
Должен быть вроде в dp. Так как в sp указываем размер текста.

BikerIndian replied:
исправил

arsensaliev commented on feature-profile-edit/src/main/res/values/themes.xml:
```diff
@@ -0,0 +1,52 @@
+    </style>
+
+    <style name="errorTextInputLayout">
+        <item name="android:textSize">12sp</item>
```
в dimens

BikerIndian replied:
перенес

BikerIndian committed:
refactor: corrections on comments

arsensaliev commented on feature-profile-edit/src/main/java/com/demal/feature_profile_edit/main/ProfileEditFragment.kt:
```diff
@@ -0,0 +1,88 @@
+
+
+
+
```
Лишние отступы.

BikerIndian replied:
fixed

BikerIndian committed:
fixed: CardView in ConstraintLayout

BikerIndian committed:
fixed: ImageView in FrameLayout

arsensaliev commented:
Исправь конфликты

BikerIndian committed:
Merge remote-tracking branch 'origin/dev' into feature/profile-edit

# Conflicts:
#	app/src/main/java/com/demal/koin/modules/ViewModelModules.kt

BikerIndian committed:
fixed: update merge
